File: N-queen-completion-problem.js
function queens(position, size){
    const to = x => [x[1] == '0'?0:size - +x[1], x[0].charCodeAt() - 97]
    const po = x => String.fromCharCode(97+x[1]) + ((size - x[0]) % 10);
    const S = JSON.stringify;
    position = to(position);
    const is_possible = (p, v) => !([...v].map(JSON.parse).some(x=>x[0]==p[0]||x[1]==p[1]||p[0]+p[1]==x[1]+x[0]||x[1]-x[0]==p[1]-p[0]))
    let visited = new Set([S(position)])
    function DFS(row, visited){
        if(visited.size == size){
            return [...visited].sort().map(x=>po(JSON.parse(x)));
        }
        
        if(row == position[0]) return DFS(row + 1, visited);
        for(let i = 0; i < size; i++){
            if(is_possible([row, i], visited)){
                let V = DFS(row + 1, new Set([...visited, S([row, i])]));
                if(V)
                    return V;
            }
        }
        return false;
    }
    return(DFS(0,visited)+'')
}

File: N-queen-completion-problem.py
def queens(position, size):
    to = lambda x: (0 if x[1]=='0' else size - int(x[1]), ord(x[0]) - 97)
    po = lambda x: f'{chr(97 + x[1])}{(size - x[0])%10}'
    position = to(position)
    is_possible=lambda p, v: not any(i[0]==p[0]or i[1]==p[1] or sum(i)==sum(p)or i[1]-i[0] == p[1] - p[0] for i in v)
    visited = set([position])
    def DFS(row, visited):
        print(len(visited), size - 1)
        if len(visited) == size:            return map(po, sorted(visited))
        if row == position[0]:              return DFS(row+1, visited)
        for i in range(size):
            if is_possible((row, i), visited):
                V = DFS(row + 1, visited | {(row, i)})
                if V:                   return V
        return False
    return ','.join([*DFS(0, visited)])
